parser-native/src/main/java/ru/urururu/sanity/cpp/NativeInstructionParser.java

```diff
@@ -0,0 +1,478 @@
+            opcodeOperatorMap.put(LLVMOpcode.LLVMSRem, BinaryExpression.Operator.Rem);
+            opcodeOperatorMap.put(LLVMOpcode.LLVMFRem, BinaryExpression.Operator.Rem);
+
+            opcodeOperatorMap.put(LLVMOpcode.LLVMAnd, BinaryExpression.Operator.And);
+            opcodeOperatorMap.put(LLVMOpcode.LLVMOr, BinaryExpression.Operator.Or);
+            opcodeOperatorMap.put(LLVMOpcode.LLVMXor, BinaryExpression.Operator.Xor);
+
+            opcodeOperatorMap.put(LLVMOpcode.LLVMShl, BinaryExpression.Operator.ShiftLeft);
+            opcodeOperatorMap.put(LLVMOpcode.LLVMLShr, BinaryExpression.Operator.ShiftRight);
+            opcodeOperatorMap.put(LLVMOpcode.LLVMAShr, BinaryExpression.Operator.ShiftRight);
+        }
+
+        @Deprecated
+        public BinaryOperationParser(LLVMOpcode opcode, BinaryExpression.Operator operator) {
+            opcodeOperatorMap = Collections.singletonMap(opcode, operator);
+        }
+
+        @Override
+        public Set<LLVMOpcode> getOpcodes() {
+            return opcodeOperatorMap.keySet();
+        }
+
+        @Override
+        public Cfe parse(NativeCfgBuildingCtx ctx, SWIGTYPE_p_LLVMOpaqueValue instruction) {
+            return createBinaryAssignment(ctx, instruction,
+                    bitreader.LLVMGetOperand(instruction, 0),
+                    opcodeOperatorMap.get(bitreader.LLVMGetInstructionOpcode(instruction)),
+                    bitreader.LLVMGetOperand(instruction, 1));
+        }
+
+        @Override
+        public RValue parseValue(NativeCfgBuildingCtx ctx, SWIGTYPE_p_LLVMOpaqueValue instruction) {
+            return ctx.getTmpVar(instruction);
+        }
+    }
+
+    private class BitCastParser extends AbstractParser {
+        @Override
+        public LLVMOpcode getOpcode() {
+            return LLVMOpcode.LLVMBitCast;
+        }
+
+        @Override
+        public Cfe parse(NativeCfgBuildingCtx ctx, SWIGTYPE_p_LLVMOpaqueValue instruction) {
+            return createBitCast(ctx, instruction, bitreader.LLVMGetOperand(instruction, 0));
+        }
+
+        @Override
+        public RValue parseValue(NativeCfgBuildingCtx ctx, SWIGTYPE_p_LLVMOpaqueValue instruction) {
+            return ctx.getTmpVar(instruction);
+        }
+
+        @Override
+        public RValue parseConst(NativeCfgBuildingCtx ctx, SWIGTYPE_p_LLVMOpaqueValue constant) {
+            return parsers.parseRValue(ctx, bitreader.LLVMGetOperand(constant, 0));
+        }
+    }
+
+    private class CastParser extends AbstractParser {
+        @Override
+        public Set<LLVMOpcode> getOpcodes() {
+            return new HashSet<>(Arrays.asList(LLVMOpcode.LLVMSExt, LLVMOpcode.LLVMZExt, LLVMOpcode.LLVMTrunc, LLVMOpcode.LLVMSIToFP));
+        }
+
+        @Override
+        public Cfe parse(NativeCfgBuildingCtx ctx, SWIGTYPE_p_LLVMOpaqueValue instruction) {
+            // todo think about source range comparison. if different - it's better to have tmp var assignment to preserve source reference.
```

Comment:
![INFO](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/severity-info.png 'Severity: INFO') Complete the task associated to this TODO comment. [![rule](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/rule.png)](https://sonarqube.com/coding_rules#rule_key=squid%3AS1135)